src/repr/src/row.rs

```diff
@@ -116,12 +116,37 @@ pub struct Row {
 impl Row {
     const SIZE: usize = CompactBytes::MAX_INLINE;
 
-    /// A variant of `Row::from_proto` that allows for reuse of internal allocs.
-    pub fn decode_from_proto(&mut self, proto: &ProtoRow) -> Result<(), String> {
+    /// A variant of `Row::from_proto` that allows for reuse of internal allocs
+    /// and validates the decoding against a provided [`RelationDesc`].
+    pub fn decode_from_proto(
+        &mut self,
+        proto: &ProtoRow,
+        desc: &RelationDesc,
+    ) -> Result<(), String> {
+        let mut col_idx = 0;
         let mut packer = self.packer();
         for d in proto.datums.iter() {
             packer.try_push_proto(d)?;
+            col_idx += 1;
+        }
+
+        // TODO(parkmycar): This is where we'll add new default values to Codec
+        // data whose upstream relation has had columns added.
+
+        // HACK(parkmycar): Only validate that the decoded Row matches the RelationDesc if it was
```

Comment:
We'll have to generalize this when we add persist support for projection pushdown, but the HACK seems fine for now

Reply:
Totally agree, in my head projection pushdown is part of or wraps a schema (I think `parquet-rs` does something similar) so ideally we'll have that info at this time.